### DISCUSSION TITLE
which router did receive the message?

### DISCUSSION THREAD

**Discussion by wspeth** [Koenkk/zigbee2mqtt#8579] [General]
hopefully not completely wrong here:
I am going to set up a larger mesh (some dozens of routers) and need to figure out which router forwarded the message of an end device (like push button or temp sensor). I could not figure out from what google rendered.. So essentially: when the device sends values and zigbee2mqtt gives those with device's network address and LQI, I also need to know the receiving router's network address and if possible the LQI on the other end. Any chance to accomplish this?

**Comment by Koenkk** [comment]
This information is not included in the message received by the coordinator.

The hops can be seen by sniffing the traffic: https://www.zigbee2mqtt.io/how_tos/how_to_sniff_zigbee_traffic.html

**Reply by wspeth** [reply]
this is bad for my project. But isn't it the linkquality a device reports refers to a router it "senses". If so couldn't it also put the router's network address into the payload? Would devices be around in the wild that do so?

**Reply by Koenkk** [reply]
No, that is against the zigbee spec and other routers won't be able to parse that message

**Reply by wspeth** [reply]
hm.. thanks for the explanation. I really appreciate your in-depth knowledge. I was mistaken assuming the device creates the "payload". it's zigbee2mqtt's responsibility, right? The device has to adhere to some profile. But couldn't the device declare the information as e.g. temperature value? Of course such trick can only work if the router's ID is known at all to the device. Feels I have do dig deeper into the matter like installing CCS and learn from sample codes. long and exhausting way. wish me luck.
